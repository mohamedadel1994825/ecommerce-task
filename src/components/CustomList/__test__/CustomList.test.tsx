import { cleanup, render } from "@testing-library/react-native";
import React from "react";
import { Text } from "react-native";
import { CustomList } from "../index";

describe("CustomList", () => {
  afterEach(cleanup);

  const data = [1, 2, 3, 4, 5];

  const renderItem = ({ item }: { item: number }) => (
    <Text testID="list-item">{item}</Text>
  );

  it("should match snapshot", () => {
    const { toJSON } = render(
      <CustomList data={data} renderItem={renderItem} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("should render correct number of items", () => {
    const { getAllByTestId } = render(
      <CustomList data={data} renderItem={renderItem} />
    );
    const listItems = getAllByTestId("list-item");
    expect(listItems.length).toBe(data.length);
  });

  it("should render separator between items", () => {
    const { getAllByTestId } = render(
      <CustomList data={data} renderItem={renderItem} />
    );
    const separators = getAllByTestId("list-item-separator");
    expect(separators.length).toBe(data.length - 1);
  });
});
