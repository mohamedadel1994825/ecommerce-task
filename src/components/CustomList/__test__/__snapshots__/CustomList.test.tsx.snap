// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomList should match snapshot 1`] = `
<View>
  <RCTScrollView
    ItemSeparatorComponent={
      {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      }
    }
    contentContainerStyle={
      {
        "paddingHorizontal": 15,
      }
    }
    data={
      [
        1,
        2,
        3,
        4,
        5,
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          testID="list-item"
        >
          1
        </Text>
        <View
          style={
            {
              "width": 15,
            }
          }
          testID="list-item-separator"
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          testID="list-item"
        >
          2
        </Text>
        <View
          style={
            {
              "width": 15,
            }
          }
          testID="list-item-separator"
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          testID="list-item"
        >
          3
        </Text>
        <View
          style={
            {
              "width": 15,
            }
          }
          testID="list-item-separator"
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          testID="list-item"
        >
          4
        </Text>
        <View
          style={
            {
              "width": 15,
            }
          }
          testID="list-item-separator"
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          testID="list-item"
        >
          5
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
